src/gpb_gen_types.erl

```diff
@@ -169,14 +169,23 @@ format_hrl_export_types(Defs, AnRes, Opts) ->
         false ->
             "";
         true ->
+            TEnv = #t_env{ 
+                type_specs = true,
+                can_do_map_presence = true,
+                mapping_and_unset =  #maps{},
+                map_key_type = atom,
+                map_type_fields = maps,
+                module = na,
+                nif = true,
+                verify_decode_required_present = true},
             iolist_to_binary(
               ["%% enumerated types\n",
                gpb_lib:nl_join([format_enum_typespec(Enum, Enumeration, AnRes)
                                 || {{enum, Enum}, Enumeration} <- Defs]),
                "\n\n",
                "%% message types\n",
                gpb_lib:nl_join(
-                 [format_record_typespec(Name, Fields, Defs, AnRes, Opts, #t_env{mapping_and_unset=records})
+                 [format_record_typespec(Name, Fields, Defs, AnRes, Opts, TEnv)
```

Comment:
I realize you did this change to fix the dialyzer issue, but I'm concerned this looks like a change in behavior here. why isn't `mapping_and_unset` field set to `records` atom?

Reply:
good point

Reply:
nice catch - missed this after pasting and filling out the record definition.